stdlib/public/Platform/android.modulemap

```diff
@@ -0,0 +1,661 @@
+//===--- android.modulemap ------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+// The module map for the Android NDK.
+// The majority of the modules in this file are prefixed
+// with an underscore, to discourage users from importing them from
+// Swift directly. Instead, users should import 'Android' or another
+// umbrella module that includes these headers.
+
+// The top level 'Android' module can be included from Swift.
+module Android [system] {
```

Comment:
I think you will have to name this differently, similar to how [it's named `SwiftGlibc` on linux](https://github.com/apple/swift/blob/78c0540b6fca098983fe6802fa0382e286586106/stdlib/public/Platform/glibc.modulemap.gyb#L21). Otherwise, the compiler can't seem to make out the difference from this clang module and the Swift module you added in #72634:
```
/data/data/com.termux/files/home/swift/stdlib/public/Platform/Android.swift:86:8: error: circular reference
84 │ public let DBL_TRUE_MIN = Double.leastNonzeroMagnitude
85 │
86 │ public let M_LN2 = Android.M_LN2
   │        │   ├─ note: through reference here
   │        │   ╰─ note: through reference here
   │        ╰─ error: circular reference
87 │ public let M_LOG10E = Android.M_LOG10E
88 │ public let M_2_SQRTPI = Android.M_2_SQRTPI

/data/data/com.termux/files/home/swift/stdlib/public/Platform/Android.swift:87:8: error: circular reference
85 │
86 │ public let M_LN2 = Android.M_LN2
87 │ public let M_LOG10E = Android.M_LOG10E
   │        │   ├─ note: through reference here
   │        │   ╰─ note: through reference here
   │        ╰─ error: circular reference
88 │ public let M_2_SQRTPI = Android.M_2_SQRTPI
89 │

/data/data/com.termux/files/home/swift/stdlib/public/Platform/Android.swift:88:8: error: circular reference
86 │ public let M_LN2 = Android.M_LN2
87 │ public let M_LOG10E = Android.M_LOG10E
88 │ public let M_2_SQRTPI = Android.M_2_SQRTPI
   │        │   ├─ note: through reference here
   │        │   ╰─ note: through reference here
   │        ╰─ error: circular reference
```
Are you able to cross-compile the stdlib for Android with these two pulls?

Reply:
Yes, @compnerd and I were able to cross compile Android stdlib correctly on Windows. I will try again though to see if I can reproduce this.

Reply:
OK, these errors are from natively compiling the March 1 trunk snapshot tag on my Android phone with these two pulls applied.

Obviously, if you plan on naming the Android libc `Bionic` now, that will require [adding `import Bionic` to a ton of compiler tests in this repo](https://github.com/apple/swift/blob/26aafe66f6eea51fc13281ebf378db49ec6aec2f/test/stdlib/POSIX.swift#L10) and external Swift packages that currently compile for Android with `import Glibc` instead.

Are you sure you want to make this libc name change on Android now? You could keep the `Glibc` name for now, but simply change this modulemap for Android, then finally flip the switch to the `Bionic` name later.